code/datums/ammo/bullet/special_ammo.dm

```diff
@@ -0,0 +1,176 @@
+	name = "holo-targeting smartgun bullet"
+	damage = 30
+///Stuff for the HRP holotargetting stacks
+	var/holo_stacks = 15
+
+/datum/ammo/bullet/smartgun/holo_target/on_hit_mob(mob/M, obj/projectile/P)
+	. = ..()
+	M.AddComponent(/datum/component/bonus_damage_stack, holo_stacks, world.time)
+
+/datum/ammo/bullet/smartgun/holo_target/ap
+	name = "armor-piercing smartgun bullet"
+	icon_state = "bullet"
+
+	accurate_range = 12
+	accuracy = HIT_ACCURACY_TIER_2
+	damage = 20
+	penetration = ARMOR_PENETRATION_TIER_8
+	damage_armor_punch = 1
+
+/datum/ammo/bullet/smartgun/m56_fpw
+	name = "\improper M56 FPW bullet"
+	icon_state = "redbullet"
+	flags_ammo_behavior = AMMO_BALLISTIC
+
+	max_range = 7
+	accuracy = HIT_ACCURACY_TIER_7
+	damage = 35
+	penetration = ARMOR_PENETRATION_TIER_1
+
+/datum/ammo/bullet/turret
+	name = "autocannon bullet"
+	icon_state = "redbullet" //Red bullets to indicate friendly fire restriction
+	flags_ammo_behavior = AMMO_BALLISTIC|AMMO_IGNORE_COVER
+
+	accurate_range = 22
+	accuracy_var_low = PROJECTILE_VARIANCE_TIER_8
+	accuracy_var_high = PROJECTILE_VARIANCE_TIER_8
+	max_range = 22
+	damage = 30
+	penetration = ARMOR_PENETRATION_TIER_7
+	damage_armor_punch = 0
+	pen_armor_punch = 0
+	shell_speed = 2*AMMO_SPEED_TIER_6
+	accuracy = HIT_ACCURACY_TIER_5
+
+/datum/ammo/bullet/turret/dumb
+	icon_state = "bullet"
+	flags_ammo_behavior = AMMO_BALLISTIC
+
+/datum/ammo/bullet/machinegun //Adding this for the MG Nests (~Art)
+	name = "machinegun bullet"
+	icon_state = "bullet" // Keeping it bog standard with the turret but allows it to be changed
+
+	accurate_range = 12
+	damage = 35
+	penetration= ARMOR_PENETRATION_TIER_10 //Bumped the penetration to serve a different role from sentries, MGs are a bit more offensive
+	accuracy = HIT_ACCURACY_TIER_3
+
+/datum/ammo/bullet/machinegun/set_bullet_traits()
+	. = ..()
+	LAZYADD(traits_to_give, list(
+		BULLET_TRAIT_ENTRY(/datum/element/bullet_trait_iff)
+	))
+
+/datum/ammo/bullet/machinegun/auto // for M2C, automatic variant for M56D, stats for bullet should always be moderately overtuned to fulfill its ultra-offense + flank-push purpose
+	name = "heavy machinegun bullet"
+
+	accurate_range = 10
+	damage =  50
+	penetration = ARMOR_PENETRATION_TIER_6
+	accuracy = -HIT_ACCURACY_TIER_2 // 75 accuracy
+	shell_speed = AMMO_SPEED_TIER_2
+	max_range = 15
+	effective_range_max = 7
+	damage_falloff = DAMAGE_FALLOFF_TIER_8
+
+/datum/ammo/bullet/machinegun/auto/set_bullet_traits()
+	return
+
+/datum/ammo/bullet/minigun
+	name = "minigun bullet"
+	headshot_state = HEADSHOT_OVERLAY_MEDIUM
+
+	accuracy = -HIT_ACCURACY_TIER_3
+	accuracy_var_low = PROJECTILE_VARIANCE_TIER_6
+	accuracy_var_high = PROJECTILE_VARIANCE_TIER_6
+	accurate_range = 12
+	damage = 35
+	penetration = ARMOR_PENETRATION_TIER_6
+
+/datum/ammo/bullet/minigun/New()
+	..()
+	if(SSticker.mode && MODE_HAS_FLAG(MODE_FACTION_CLASH))
```

Comment:
This looks like all sorts of jank. I appreciate that this is an uplift from an existing piece of code but maybe we can raise a ticket to clean this up?